internal/scraper/scraper.go

```diff
@@ -501,11 +501,23 @@ func (s Scraper) collectData(ctx context.Context, t time.Time) (*probeData, time
 
 	sl := kitlog.With(bl, loggerLabels...)
 
+	var timeout time.Duration
+	switch s.CheckType() {
+	case sm.CheckTypeMultiHttp, sm.CheckTypeScripted, sm.CheckTypeBrowser:
+		// k6-backed checks have a more sophisticated handling of timeouts, where the k6 runner receives the check
+		// timeout from the specification and may implement a retrying mechanism. For this reason, we set the prober
+		// timeout to the check frequency instead, which is the hard limit at which no further retries are possible as
+		// the next execution is due.
+		timeout = time.Duration(s.check.Frequency) * time.Millisecond
+	default:
+		timeout = time.Duration(s.check.Timeout) * time.Millisecond
+	}
+
 	success, mfs, err := getProbeMetrics(
 		ctx,
 		s.prober,
 		target,
-		time.Duration(s.check.Timeout)*time.Millisecond,
+		timeout,
```

Comment:
Adding retries may impact the probe duration metric in https://github.com/grafana/synthetic-monitoring-agent/blob/f193b2e1d07c379d0d6737795f00f4954eb02fe7/internal/scraper/scraper.go#L650

However, there's an argument to be made that with external runners with queues and network latency, this metric does not make sense for k6-backed checks already. I think we can address that on a separate PR.

Reply:
There's a hack that pulls out the duration from the emitted metrics. See line 530.